soul-client/soul-client-core/src/main/java/org/dromara/soul/client/core/disruptor/subcriber/SoulClientMetadataExecutorSubscriber.java

```diff
@@ -50,6 +53,17 @@ public DataType getType() {
     @Override
     public void executor(final Collection<MetaDataRegisterDTO> metaDataRegisterDTOList) {
         for (MetaDataRegisterDTO metaDataRegisterDTO : metaDataRegisterDTOList) {
+            while (true) {
+                try (Socket socket = new Socket(metaDataRegisterDTO.getHost(), metaDataRegisterDTO.getPort())) {
+                    break;
+                } catch (IOException e) {
+                    try {
+                        TimeUnit.MILLISECONDS.sleep(100);
+                    } catch (InterruptedException ex) {
```

Comment:
Can we add the maximum number of retries to prevent an endless loop ?

Reply:
I don't think there will be an endless loop. If the socket cannot be successfully created, the service did not start successfully and will be terminated.